### ISSUE TITLE
Clarification of `sort-imports` rule with destructuring imports

### ISSUE THREAD

**Issue by wyze** [eslint/eslint#5311]
ESLint version `2.1.0`.

I think any destructuring import should be grouped together when using `sort-imports` rule.

Right now, this is valid with default options: 

```js
import { Router, browserHistory as history } from 'react-router'
import React from 'react'
import { render } from 'react-dom'
import routes from './routes'
```

Where as, it is easier to read if everything that started with a `{` were grouped together, like so:

```js
import { Router, browserHistory as history } from 'react-router'
import { render } from 'react-dom'
import React from 'react'
import routes from './routes'
```

While technically, it is a single import, I find it looks out of place. So, I am not sure if this would be considered a bug, or a feature request?

Sorry about the title, not sure how to describe this well. Feel free to adjust the title.

**Comment by nzakas** [comment]
Have you tried changing the default options? http://eslint.org/docs/rules/sort-imports.html#membersyntaxsortorder

**Comment by michaelficarra** [comment]
Duplicate of #5305?

**Comment by ilyavolodin** [comment]
I think it's slightly different then #5305 because it deals with multiple import statements. However, @wyze would you mind testing the same code with the latest master? We just merged a fix for import sorting, and it might resolve your issue too.

**Comment by wyze** [comment]
@nzakas I haven't changed the default options, as the destructuring of one import is considered a `single`, I don't see how the options would solve that.

@ilyavolodin I tried the latest master and it reports the same as before.

**Comment by ilyavolodin** [comment]
Looking at the available options, it seems that this would be a new feature request. Right now we are sorting based on the first member.

**Comment by wyze** [comment]
I would be willing to submit a PR for this, given some direction of course. If it would be another option that could be passed to the rule, and naming around that option.

**Comment by nzakas** [comment]
What would that option be? Keep in mind, this is not destructuring, it's individual binding import. Looks similar, behaves differently.

**Comment by wyze** [comment]
I went ahead and implemented this in my own plugin as I was uncertain if it would get merged into core. I just wanted to share the diff, which isn't too bad. I am not sure if this could be done with an option or not, but would be cool to see this in core.

[Line 19](https://github.com/eslint/eslint/blob/master/lib/rules/sort-imports.js#L19)
```diff
-  memberSyntaxSortOrder = configuration.memberSyntaxSortOrder || ["none", "all", "multiple", "single"],
+  memberSyntaxSortOrder = configuration.memberSyntaxSortOrder || ["none", "all", "named", "default"],
```

[Lines 38-41](https://github.com/eslint/eslint/blob/master/lib/rules/sort-imports.js#L38-L41)
```diff
-  } else if (node.specifiers.length === 1) {
-    return "single";
-  } else {
-    return "multiple";
+  } else if (node.specifiers[0].type === "ImportDefaultSpecifier") {
+    return "default";
+  } else {
+    return "named";
```

[Line 153](https://github.com/eslint/eslint/blob/master/lib/rules/sort-imports.js#L153)
```diff
-  "enum": ["none", "all", "multiple", "single"]
+  "enum": ["none", "all", "named", "default"]
```

You can see my full implementation over here: https://github.com/wyze/eslint-plugin-wyze

Curious on your thoughts and feedback.

**Comment by nzakas** [comment]
@eslint/eslint-team does anyone want to champion this?

**Comment by AnthonyAstige** [comment]
I found the current default sort order confusing. ie; it was weird to me that `import { ModuleName } from ...` is grouped with `import Job from ...`. In sorting my imports it made me think something was wrong with the rule before I read this issue.

**Comment by nzakas** [comment]
No team members stepped forward to champion this change, so closing.
